Neos.Neos/Classes/Domain/Service/WorkspacePublishingService.php

```diff
@@ -255,19 +276,15 @@ private function discardNodes(
         );
     }
 
+    /**
+     * @throws WorkspaceRebaseFailed is thrown if the workspace was outdated and an automatic rebase failed due to conflicts.
+     * No changes would be published for this case.
+     */
     private function publishNodes(
         ContentRepository $contentRepository,
         WorkspaceName $workspaceName,
         NodeIdsToPublishOrDiscard $nodeIdsToPublish
     ): void {
-        /**
-         * TODO: only rebase if necessary!
-         * Also, isn't this already included in @see WorkspaceCommandHandler::handlePublishIndividualNodesFromWorkspace ?
-         */
-        $contentRepository->handle(
-            RebaseWorkspace::create($workspaceName)
-        );
-
         $contentRepository->handle(
             PublishIndividualNodesFromWorkspace::create(
```

Comment:
We throw now the `WorkspaceRebaseFailed` in case one of the commands could not be handled, previously we would just throw the raw exception in `PublishIndividualNodesFromWorkspace` instead.

The `WorkspaceRebaseFailed` was only due to the now obsolete `RebaseWorkspace` here which is required for the ui to react accordingly: https://github.com/neos/neos-ui/pull/3769

see https://github.com/neos/neos-development-collection/pull/5301/commits/8ad0a3e5c5f968f8b8b1fda83b3df6710834b95b